.github/workflows/bundle-audit.yml

```diff
@@ -15,4 +15,6 @@ jobs:
           ruby-version: '2.7'
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
       - name: Bundle audit
-        run: bundle audit --update
+        run: |
+          gem install bundle-audit
```

Comment:
Why not a separate step?

Reply:
No idea, what's the advantage/disadvantage?

Reply:
I was thinking whether or not bundler-audit should be managed by Gemfile.lock or not and I'm going to leave a note for you to decide.

Clockwork uses Gemfile.lock to manage bundler-audit version, this has the advantage that we control when we upgrade to the newer version. But the downside is that we stay on the old version unless we upgrade.

On the other hand here we're moving away from this approach and are installing bundle-audit on our own. I would assume that means the newest version will be installed, which is great. Unless it breaks our build. So we're introducing some volatility into the build. Still, we use other volatile sources too, so I'm not worried about flakiness that much. Our builds are generally flaky due to using external repositories, etc.

The question is which approach is generally perceived by as us better and should we use only one strategy moving forward? Leaving that up to you. I shared my considerations here.

Reply:
Valid concerns.

My thoughts: who is triggering the build. If it's human-triggered (i.e. by pushing to the branch), then I'd say the stability is much more important (i.e. "if commit X was yesterday working, tomorrow commit X should also work"). I hate when during my daily work I have to fix irrelevant things in build infrastructure just to deliver my hotfix/feature to the customer.

However, if the build was/is triggered differently... :thinking:  
...in our case, the bundler-audit builds should fail as soon as one dependency becomes insecure. So, ideally, it's the action originating outside of our control (some random researcher marking a random component as vulnerable).

Honestly, I dunno. It's not that the bundler-audit builds are kind of gatekeeper builds. Their failure doesn't prevent us from hotfixing production. They are more like monitoring builds, telling us that something is wrong with our production. Signaling some problem to address, similar to i.e. 'memory out of space', or "too many postgres connections". 

----

Let's try with the approach of installing the newest bundle-audit version and see if it breaks our build later. If it breaks, then let's go back to the drawing board and decide if:
* we want stability from the audit-builds (I don't, cause they are not 'gatekeeping' builds)
* we want to run audits on cutting edge version (i.e. because security is much more important to us than builds stability)

Reply:
@skv-headless - I'll move forward with this PR, but if you have input (i.e. you see the value to have consistent approach regarding to using tools like "bundler audit" across all our codebase), then I'm/we're listening.